tests/src/3d/testqgs3dutils.cpp

```diff
@@ -516,6 +519,88 @@ void TestQgs3DUtils::testScreenPointToMapCoordinates()
   QGSCOMPARENEAR( mapPoint.z(), -252, 2 );
 }
 
+void TestQgs3DUtils::testLineSegmentToClippingPlanes()
+{
+  const QgsPointXY point1( 20, 20 );
+  const QgsPointXY point2( 50, 50 );
+  const QgsVector3D midPoint( point2.x() - point1.x(), point2.y() - point1.y(), 0 );
+  QList<QVector4D> clippingPlanes = Qgs3DUtils::lineSegmentToClippingPlanes( point1, point2, 10, QgsVector3D( 0, 0, 0 ) );
+  for ( int i = 0; i < clippingPlanes.size(); i++ )
+  {
+    QgsVector3D planePoint(
+      i % 2 ? midPoint.x() : ( -clippingPlanes.at( i ).y() * midPoint.y() - clippingPlanes.at( i ).w() ) / clippingPlanes.at( i ).x(),
+      i % 2 ? ( -clippingPlanes.at( i ).x() * midPoint.x() - clippingPlanes.at( i ).w() ) / clippingPlanes.at( i ).y() : midPoint.y(),
+      0
+    );
+    const double distance = QgsVector3D::dotProduct( midPoint - planePoint, clippingPlanes.at( i ).toVector3D() );
+    // verify all normals are pointing inside the rectangle
+    QVERIFY( distance > 0 );
+  }
+
+  QVERIFY( QVector4D::dotProduct( clippingPlanes.at( 0 ).toVector3D(), clippingPlanes.at( 1 ).toVector3D() ) == 0 );
+  QVERIFY( QVector4D::dotProduct( clippingPlanes.at( 1 ).toVector3D(), clippingPlanes.at( 2 ).toVector3D() ) == 0 );
+  QVERIFY( QVector4D::dotProduct( clippingPlanes.at( 2 ).toVector3D(), clippingPlanes.at( 3 ).toVector3D() ) == 0 );
+  QVERIFY( QVector4D::dotProduct( clippingPlanes.at( 3 ).toVector3D(), clippingPlanes.at( 0 ).toVector3D() ) == 0 );
```

Comment:
I would probably remove these checks - there's a good chance they will break due to some numerical issues and return some small number instead